apps/dashboard/app/(app)/authorization/roles/components/upsert-role/components/assign-key/create-key-options.tsx

```diff
@@ -0,0 +1,140 @@
+import { StatusBadge } from "@/app/(app)/apis/[apiId]/settings/components/status-badge";
+import { Badge } from "@/components/ui/badge";
+import { Key2, Lock } from "@unkey/icons";
+import { Button, Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from "@unkey/ui";
+
+type Key = {
+  id: string;
+  name: string | null;
+  roles: {
+    id: string;
+    name: string;
+  }[];
+};
+
+type KeySelectorProps = {
+  keys: Key[];
+  hasNextPage?: boolean;
+  isFetchingNextPage: boolean;
+  roleId?: string;
+  loadMore: () => void;
+};
+
+export function createKeyOptions({
+  keys,
+  hasNextPage,
+  isFetchingNextPage,
+  roleId,
+  loadMore,
+}: KeySelectorProps) {
+  const options = keys.map((key) => ({
+    label: (
+      <TooltipProvider>
+        <Tooltip>
+          <TooltipTrigger asChild>
+            <div className="flex w-full text-accent-8 text-xs gap-4 py-0.5 items-center group flex-row">
+              <div className="border rounded-full flex items-center justify-center border-grayA-6 size-5">
+                <Key2 size="sm-regular" className="text-grayA-11" />
+              </div>
+              <div className="flex gap-1 flex-col truncate">
+                <div className="flex gap-2 items-center">
+                  <span className="font-medium text-accent-12 text-left">
+                    {key.name || "Unnamed Key"}
+                  </span>
+                  {key.roles.find((item) => item.id === roleId) && (
+                    <StatusBadge
+                      variant="locked"
+                      text="Already assigned"
+                      icon={<Lock size="sm-thin" />}
+                    />
+                  )}
+                </div>
+                <span className="text-accent-9 text-xs">
+                  {key.id.length > 15 ? `${key.id.slice(0, 8)}...${key.id.slice(-4)}` : key.id}
```

Comment:
booo magic strings. what does 15 and 8 and -4 represent? why these numbers? constants would be great here:

```
const ID_TRUNCATE_LENGTH = 15;
const ID_TRUNCATE_START = 8;
const ID_TRUNCATE_END = 4;
```

Reply:
I don't think that thing above will make a huge difference. its just trimming the key maybe we can create a helper func for that.

Reply:
⬆️ We have that mostly in API Key related pages

Reply:
Yeah they are exactly like that